src/AddIns/Uno.WinUI.Graphics2DSK/SKCanvasElement.cs

```diff
@@ -12,12 +17,46 @@ namespace Uno.WinUI.Graphics2DSK;
 /// <remarks>This is only available on skia-based targets.</remarks>
 public abstract class SKCanvasElement : FrameworkElement
 {
-	private protected override ContainerVisual CreateElementVisual() => new SKCanvasVisual(this, Compositor.GetSharedCompositor());
+#if !WINAPPSDK
+	private SKCanvasVisualBase? _skCanvasVisual;
+
+	private protected override ContainerVisual CreateElementVisual()
+	{
+		if (ApiExtensibility.CreateInstance<SKCanvasVisualBaseFactory>(this, out var factory))
+		{
+			return _skCanvasVisual = factory.CreateInstance((o, size) => RenderOverride((SKCanvas)o, size), Compositor.GetSharedCompositor());
+		}
+		else
+		{
+			throw new InvalidOperationException($"Failed to create an instance of {nameof(SKCanvasVisualBase)}");
+		}
+	}
+#endif
+
+	protected SKCanvasElement()
+	{
+#if !WINAPPSDK
+		if (!ApiExtensibility.IsRegistered<SKCanvasVisualBaseFactory>())
+#endif
+		{
+			throw new PlatformNotSupportedException($"This platform does not support {nameof(SKCanvasElement)}");
+		}
+	}
+
+#if WINAPPSDK
+	public static bool IsSupportedOnCurrentPlatform() => false;
+#else
+	public static bool IsSupportedOnCurrentPlatform() => ApiExtensibility.IsRegistered<SKCanvasVisualBaseFactory>();
+#endif
 
 	/// <summary>
 	/// Invalidates the element and triggers a redraw.
 	/// </summary>
-	public void Invalidate() => _visual.Compositor.InvalidateRender(_visual);
+#if WINAPPSDK
+	public void Invalidate() { }
+#else
+	public void Invalidate() => _skCanvasVisual?.Invalidate();
+#endif
```

Comment:
I'm surprised that we need this check, would should not make this project available to netX.0-windows (only through netX.0), is it because of the winappsdk samples app?

Reply:
We need to reference WinAppSdk's packages conditionally when not compiling for Uno, so we need the different TFM to differentiate the target (and I'm not even sure if the project would build with those packages on net9.0).